includes/migrate.form.inc

```diff
@@ -56,6 +57,9 @@ function islandora_basic_collection_migrate_item_form_validate($form, &$form_sta
   if (!$is_a_collection) {
     form_set_error('new_collection_name', t('Not a valid collection'));
   }
+  if ($object === $new_collection) {
```

Comment:
Just a comment, but in general comparing strings is way faster than comparing objects. So maybe it would be better to compare PIDs of both before all the object loading actually happens, as soon as the function is invoked, at line 47. Then only, if they are different, continue loading objects and execute logic. Same happens further down at https://github.com/Islandora/islandora_solution_pack_collection/pull/209/files#diff-e98b2d3e00f27df311198dfea43ad7d7R56

Reply:
Thanks, I've made the change